hooks/useDebounce.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useRef } from "react";
+/**
+ * 함수와 딜레이시간, 디바운싱 방식의 옵션을 매개변수로 받아서 디바운싱이 적용된 함수를 반환합니다.
+ * @param func 디바운싱을 걸 함수
+ * @param delay 딜레이 시간
+ * @param isLeading 디바운싱 방식, 기본값 false (true이면 Leading Edge 방식, 이벤트 시작시 바로 함수실행)
+ * @returns 디바운싱 적용 함수 반환
+ */
+export default function useDebounce(
+  func: (...args: any[]) => void,
+  delay: number,
+  isLeading: boolean = false,
+) {
+  //setTimeout or null ref
+  const timeoutId = useRef<NodeJS.Timeout | null>(null);
+
+  const debouncedFunction = () => {
+    //setTimeout이 들어있는 상태이면 클리어하고 리셋
+    // 사용자가 딜레이중에 함수를 또 실행시켰다는걸 의미
+    if (timeoutId.current) {
+      clearTimeout(timeoutId.current);
+    }
+
+    //isLeading 방식이면 여기서 함수실행시킴
+    if (isLeading && !timeoutId.current) {
+      func();
+    }
+
+    //함수 실행시키고 나서 setTimeout 넣어줌
+    //딜레이 뒤에 아래함수 수행
+    timeoutId.current = setTimeout(() => {
+      //trailing 방식이면 여기서 실행
+      if (!isLeading) {
+        func();
+      }
+      //실행끝나면 null 넣어줌 (리셋)
+      timeoutId.current = null;
+    }, delay);
+  };
+
+  useEffect(() => {
+    return () => {
+      if (timeoutId.current) {
+        clearTimeout(timeoutId.current);
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
```suggestion
    // eslint-disable-next-line react-hooks/exhaustive-deps
```
이거 룰 안막으면 에러가 생기나요?? 에러가 아닌 단순 경고이고, 의도하신 부분이면 룰 안막아도 될 것 같긴해요!

Reply:
의도한 부분인데 저번에 남주님이 디펜던시 비워놓으면 린트 걸린다고하셔서 여기선 막아놓았습니다

Reply:
린트 걸린다고 하시는게 에러인가요? 경고인가요??
에러면 빌드가 안되서 룰 막는게 맞는 것 같은데, 경고이고 재영님께서 의도적으로 디펜더시 비워놓으신거면 룰 안막아도 될 것 같다는 의미였습니다!

Reply:
여기서 린트지우고 빌드해봤는데 에러나 경고가없네요..? 아마 useEffect 안에서 선언된게 없어서 그런거같아요 수정했습니다